src/Microsoft.OData.Client/Serialization/HttpClientRequestMessage.cs

```diff
@@ -0,0 +1,426 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="HttpClientRequestMessage.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.Client.Serialization
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.OData.Client;
+    using System.IO;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Threading.Tasks;
+    using Microsoft.OData;
+
+    /// <summary>
+    /// HttpClient based implementation of DataServiceClientRequestMessage.
+    /// </summary>
+    public class HttpClientRequestMessage : DataServiceClientRequestMessage, IDisposable

```

Comment:
@paulodero and @ElizabethOkerio shouldn't we retain the Web class and mark it as deprecated as with this change it could break multiple people. Probably having a class with the same name but extend the new class would be a good way to keep some backwards compatibility till we get to a major release.

Reply:
fixed